### PR TITLE
HHH-11194 - Add setting to enable legacy 4.3.x LimitHandler behavior.

### PR DESCRIPTION
**Pull request by Naros**
https://hibernate.atlassian.net/browse/HHH-11194

### CODE REVIEW

**Comment by Naros** [comment]
@sebersole The only real motive I had was to encapsulate the if-check behavior.  I can most definitely change the PR to avoid the delegation/indirection.

File: hibernate-core/src/main/java/org/hibernate/dialect/Dialect.java [hunk 1]
```diff
@@ -307,7 +310,7 @@ public String toString() {
 	 * @param serviceRegistry The service registry
 	 */
 	public void contributeTypes(TypeContributions typeContributions, ServiceRegistry serviceRegistry) {
-		// by default, nothing to do
+		resolveLegacyLimitHandlerBehavior( serviceRegistry );
```
Comment (vladmihalcea) [inline hibernate-core/src/main/java/org/hibernate/dialect/Dialect.java]: Shouldn't we have the resolveLegacyLimitHandlerBehavior method called by `MetadataBuildingProcess` just like it does for `contributeTypes`? It's a little bit hacky that we resolve the limit handler during type contributions.

**Comment by vladmihalcea** [comment]
@Naros I send you [this Pull Request](https://github.com/hibernate/hibernate-orm/pull/1650) which fixes the SQL Server test and changes the LimitHandler logic for the SQLServer2005 and Ingres9 Dialects. Let me know if it's what you intended too.

**Comment by vladmihalcea** [comment]
Applied upstream.

**Pull request closed without merging**